src/test/java/com/google/cloud/compute/v1/it/BaseTest.java

```diff
@@ -120,11 +120,28 @@ protected static Operation waitForOperation(Operation operation) {
     return completedOperation;
   }
 
-  static void cleanUpNetworks() throws IOException {
+  static void cleanUpNetwork(Network network) throws IOException {
     FirewallSettings firewallSettings =
         FirewallSettings.newBuilder().setCredentialsProvider(credentialsProvider).build();
     FirewallClient firewallClient = FirewallClient.create(firewallSettings);
 
+    // TODO: switch to firewall list filter
+    List<Firewall> firewalls =
+        Lists.newArrayList(firewallClient.listFirewalls(PROJECT_NAME).iterateAll());
+    for (Firewall firewall : firewalls) {
+      if (firewall.getNetwork().equals(network.getSelfLink())) {
+        waitForOperation(firewallClient.deleteFirewall(firewall.getSelfLink()));
+      }
+    }
+
+    NetworkSettings networkSettings =
```

Comment:
how come this logic moved from setUp to cleanup?

Reply:
In `ITNetworkTest`, the network client is created at setup time and is reused throughout the test. In the `ITFirewallTest` it was used to create a separate test network for use by the firewalls.

We could re-use the same `NetworkClient` from the test and pass both it and the `FirewallClient` to the helper functions, but it's doesn't seem like too big of a deal.